### PR TITLE
Deprecated doc

### PR DESCRIPTION
**Pull request by glicht**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/31144

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

### CODE REVIEW

File: content-repo/extra-docs/articles/deprecated.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+id: deprecated
+title: Deprecated Integrations
+description: Support dates and End of Life notices for Deprecated Integrations. 
+---
+For various reasons, Integrations reach a deprecated status. In most cases, the reason an Integration is deprecated is that a new version replaces the deprecated Integration. For Cortex XSOAR supported Integrations, the deprecation process involves the Integration entering a **Maintenance Mode** period and finally an end-of-life declaration. Cortex XSOAR tries to make this process as seamless as possible for our customers and partners while providing visibility into what can be expected during this process.
```
Comment (Itay4) [inline content-repo/extra-docs/articles/deprecated.md:6]:
```suggestion
For various reasons, Integrations reach a deprecated status. In most cases, the reason an Integration is deprecated is that a new version replaces the deprecated Integration. For Cortex XSOAR supported Integrations, the deprecation process involves the Integration entering a **Maintenance Mode** period and finally an **End-Of-Life** declaration. Cortex XSOAR tries to make this process as seamless as possible for our customers and partners while providing visibility into what can be expected during this process.
```

File: content-repo/extra-docs/articles/deprecated_info.json [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+      "id": "MISP",
+      "note": "Use the MISP v2 integration instead."
+    },
+    {
+      "id": "OPSWAT-Metadefender",
+      "note": "Use the OPSWAT-Metadefender v2 integration instead."
+    },
+    {
+      "id": "LightCyber Magna",
+      "note": "Product is has been replaced by Cortex XDR."
```
Comment (Itay4) [inline content-repo/extra-docs/articles/deprecated_info.json:97]:
```suggestion
      "note": "Product has been replaced by Cortex XDR."
```

File: content-repo/extra-docs/articles/deprecated_info.json [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+      "id": "Proofpoint TAP",
+      "note": "Use the Proofpoint TAP v2 integration instead."
+    },
+    {
+      "id": "SafeBreach",
+      "note": "Use the SafeBreach v2 integration instead."
+    },
+    {
+      "id": "Secdo",
+      "note": "Product is EOL. Product has been replaced by Cortex XDR."
```
Comment (Itay4) [inline content-repo/extra-docs/articles/deprecated_info.json:117]:
```suggestion
      "note": "Product has been replaced by Cortex XDR."
```

**Comment by Itay4** [comment]
image

the notes of these 2 are not consistent with the others

File: content-repo/gendocs.py [hunk 4]
```diff
@@ -457,6 +485,120 @@ def insert_approved_tags_and_usecases():
...
+
+
+def merge_deprecated_info(deprecated_list: List[DeprecatedInfo], deperecated_info_file: str):
+    with open(deperecated_info_file, "rt") as f:
+        to_merge_list: List[DeprecatedInfo] = json.load(f)['integrations']
+    to_merge_map = {i['id']: i for i in to_merge_list}
+    merged_list: List[DeprecatedInfo] = []
+    for d in deprecated_list:
+        if d['id'] in to_merge_map:
+            d = {**d, **to_merge_map[d['id']]}  # type: ignore
```
Comment (Itay4) [inline content-repo/gendocs.py:559]: please add error code here

**Comment by glicht** [comment]
> image
> dont think the description is meant to be here

Fixed this one.

**Comment by glicht** [comment]
> image
> image
> image
> 
> the notes of these are not consistent with the others

I am ok with these. We can fix this in the content side. We can discuss here: https://github.com/demisto/etc/issues/32712

File: content-repo/gendocs.py (hunk 4 above)
Comment (glicht) [reply content-repo/gendocs.py:563]: It is a known limitation of mypy when merging dicts. https://github.com/python/mypy/issues/4122. Will add the error code though for completeness.

**Review by Itay4 — APPROVED** [review]

**Review by anara123 — APPROVED** [review]

**Pull request merged**